datafusion/core/src/physical_plan/memory.rs

```diff
@@ -144,6 +146,25 @@ impl MemoryExec {
             schema,
             projected_schema,
             projection,
+            sort_information: None,
+        })
+    }
+    /// Create a new execution plan for reading in-memory record batches
+    /// The provided `schema` should not have the projection applied. Also, you can specify sort
+    /// information on PhysicalExprs.
+    pub fn try_new_with_sort_information(
```

Comment:
FWIW another pattern that might be nice would be a builder style:

```rust
    pub fn with_sort_information(mut self, sort_information: Option<Vec<PhysicalSortExpr>>) -> Self {
  ...
}
```

so that then the creation looks like 
```rust
let exec = MemExec::try_new(partitions, schema, projection)?
  .with_sort_information(sort_information)
```

Reply:
👍

Reply:
Revamped.